src/journal/journal-file.c

```diff
@@ -1076,23 +1076,24 @@ static int journal_file_append_data(
         o->data.hash = htole64(hash);
 
 #if defined(HAVE_XZ) || defined(HAVE_LZ4)
-        if (f->compress_xz &&
-            size >= COMPRESSION_SIZE_THRESHOLD) {
+        if (JOURNAL_FILE_COMPRESS(f) && size >= COMPRESSION_SIZE_THRESHOLD) {
                 size_t rsize = 0;
 
                 compression = compress_blob(data, size, o->data.payload, &rsize);
 
-                if (compression) {
+                if (compression >= 0) {
```

Comment:
Well, it's customary in systemd to indicate success with return values >= 0, and errors with return values < 0... Now, it's true that the two OBJECT_COMPRESSED_XYZ values are actually != 0, so we could write > 0, but I mean they actually aren't 3 or 4 either, so I'd rather not get into comparing with too specific values, but stick to the simple decision between "is this an error" or "is this not an error"... Hope that makes some sense...